### DISCUSSION TITLE
Trigger fires on game load and i'm not sure why

### DISCUSSION THREAD

**Discussion by anthonySchafrik** [excaliburjs/Excalibur#2810] [Q&A] [answered]
Hey i'm having a hard time trying to understand the docs around triggers. I'm trying to make a trigger so when the player walks on to a tile it will trigger an event to switch to a  new map. The trigger is firing off when the game first loads. I have this trigger and im adding to the game engine.

```
export const doorTrigger = new Trigger({
  width: 16,
  height: 16,
  pos: vec(1208, 136),
  repeat: -1,
  action: () => {
    console.log('triggered');
  },
});

game.currentScene.add(doorTrigger);
```

**Comment by anthonySchafrik** [comment]
Ah if you give your trigger a target actor it looks to fix my issue

**Comment by eonarheim** [comment, marked as answer]
Hi @anthonySchafrik, you are correct the `target` will make it so the trigger only fires on that actor. I've updated the docs with more details, they were pretty lacking... New page is up https://excaliburjs.com/docs/triggers/

One alternative to setting a target, is specifying a filter function. The filter function is passed any actor that intersects and returning true determines if want to trigger on that actor.

```typescript
const player = new ex.Actor({
  name: 'player',
  pos: ex.vec(0, 100),
  width: 16,
  height: 16,
  collisionType: ex.CollisionType.Active,
  color: ex.Color.Red
});

player.vel.x = 10;

export const doorTrigger = new ex.Trigger({
  width: 16,
  height: 16,
  pos: ex.vec(100, 100),
  repeat: -1,
  filter: (actor) => {
    return actor.name === 'player';
  },
  action: () => {
    console.log('triggered');
  },
});
```
You can also use collision groups instead of a `filter` as well

```typescript
const PlayerGroup = ex.CollisionGroupManager.create('player');

const player = new ex.Actor({
  name: 'player',
  pos: ex.vec(0, 100),
  width: 16,
  height: 16,
  collisionType: ex.CollisionType.Active,
  collisionGroup: PlayerGroup,
  color: ex.Color.Red
});

player.vel.x = 10;

export const doorTrigger = new ex.Trigger({
  width: 16,
  height: 16,
  pos: ex.vec(100, 100),
  repeat: -1,
  action: () => {
    console.log('triggered');
  },
});
doorTrigger.body.group = ex.CollisionGroup.collidesWith([PlayerGroup]);

```

**Reply by anthonySchafrik** [reply]
Ah I see thank you for the info this is really helpful.
